.github/workflows/release.yml

```diff
@@ -0,0 +1,45 @@
+name: Release
+
+on:
+  push:
+    tags:
+      - v*.*.*
+
+jobs:
+  dist:
+    # Build the Python SDist and wheel, performs metadata and readme linting
+    name: Build distribution
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
```

Comment:
If the docs are correct on this, you can remove this.

"unless the latest commit is the version tag."
Ref: https://github.com/hynek/build-and-inspect-python-package?tab=readme-ov-file#usage